mzLib/Proteomics/RetentionTimePrediction/ChronologerModel/ChronologerEstimator.cs

```diff
@@ -0,0 +1,344 @@
+
+            var predictionHolder = new List<float>();
+            foreach (var batch in dataLoader)
+            {
+                var output = ChronologerModel.Predict(torch.vstack(batch).to(device));
+                predictionHolder.AddRange(output.to(DeviceType.CPU).data<float>());
+                output.Dispose();
+            }
+
+            Parallel.For(0, preds.Length, outputIndex =>
+            {
+                preds[outputIndex] = predictionHolder[outputIndex];
+            });
+
+            //change to -1 if same index in compatibleTracker is false, else leave as is
+            //predictions = preds.SelectMany(x => x).ToArray();
+            // return vstacked tensors as a matrix => float?[]
+
+            for (var predictionsIndex = 0; predictionsIndex < predictions.Length; predictionsIndex++)
+            {
+                if (compatibleTracker[predictionsIndex])
+                {
+                    predictions[predictionsIndex] = predictionHolder[predictionsIndex];
+                    continue;
+                }
+
+                predictions[predictionsIndex] = -1;
+            }
+
+
+            return predictions;
+        }
+        /// <summary>
+        /// Takes the base sequence and the full peptide sequence and returns a tensor for the Chronologer model.
+        /// The base sequence is the sequence without modifications and the full peptide sequence is the sequence with modifications.
+        /// The model is intended to be used with sequences of length 50 or less, and only supports modifications present in the Chronologer dictionary.
+        /// Sequences not supported by chronologer will return null.
+        /// </summary>
+        /// <param name="baseSequence"></param>
+        /// <param name="fullSequence"></param>
+        /// <returns></returns>
+        private static (torch.Tensor, bool) Tensorize(string baseSequence, string fullSequence, out bool chronologerCompatible)
+        {
+            // Chronologer does not support metals
+            if (fullSequence.Contains("Metal") || baseSequence.Contains("U") || baseSequence.Length > 50)
+            {
+                chronologerCompatible = false;
+                return (torch.zeros(1, 52, torch.ScalarType.Int64), chronologerCompatible);
+            }
+
+            var fullSeq = fullSequence.Split(new[] { '[', ']' })
+                .Where(x => !x.Equals("")).ToArray();
+
+
+            //section to remove type of mod from the sequence
+            //e.g. Common Fixed:Carbamidomethyl and only stay with the target aa after the mod
+            for (int i = 0; i < fullSeq.Length; i++)
+            {
+                if (fullSeq[i].Contains(" "))
+                {
+                    var tempSeq = fullSeq[i].Split(':');
+                    fullSeq[i] = tempSeq[1];
+                }
+            }
+
+            if (baseSequence.Length <= 50) //Chronologer only takes sequences of length 50 or less
+            {
```

Comment:
If statement is unnecessary due to above check. that will give you coverage of lines 222-223